FreeRTOS-Plus/Demo/AWS/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/shadow_demo_helpers.c

```diff
@@ -0,0 +1,933 @@
+                    xMQTTStatus ) );
+        returnStatus = pdFAIL;
+    }
+    else
+    {
+        LogInfo( ( "SUBSCRIBE topic %.*s to broker.\n\n",
+                   usTopicFilterLength,
+                   pcTopicFilter ) );
+
+        /* Process incoming packet from the broker. Acknowledgment for subscription
+         * ( SUBACK ) will be received here. However after sending the subscribe, the
+         * client may receive a publish before it receives a subscribe ack. Since this
+         * demo is subscribing to the topic to which no one is publishing, probability
+         * of receiving publish message before subscribe ack is zero; but application
+         * must be ready to receive any packet. This demo uses MQTT_ProcessLoop to
+         * receive packet from network. */
+        xMQTTStatus = MQTT_ProcessLoop( pxMqttContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+        if( xMQTTStatus != MQTTSuccess )
+        {
+            returnStatus = pdFAIL;
+            LogError( ( "MQTT_ProcessLoop returned with status = %u.",
+                        xMQTTStatus ) );
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+BaseType_t xUnsubscribeFromTopic( const char * pcTopicFilter,
+                                  uint16_t usTopicFilterLength )
+{
+    BaseType_t returnStatus = pdTRUE;
+    MQTTStatus_t xMQTTStatus;
+    MQTTContext_t * pxMqttContext = &mqttContext;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    configASSERT( pxMqttContext != NULL );
+    configASSERT( pcTopicFilter != NULL );
+    configASSERT( usTopicFilterLength > 0 );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* This example subscribes to only one topic and uses QOS1. */
+    pSubscriptionList[ 0 ].qos = MQTTQoS1;
+    pSubscriptionList[ 0 ].pTopicFilter = pcTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = usTopicFilterLength;
+
+    /* Generate packet identifier for the UNSUBSCRIBE packet. */
+    globalUnsubscribePacketIdentifier = MQTT_GetPacketId( pxMqttContext );
+
+    /* Send UNSUBSCRIBE packet. */
+    xMQTTStatus = MQTT_Unsubscribe( pxMqttContext,
+                                    pSubscriptionList,
+                                    sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                    globalUnsubscribePacketIdentifier );
+
+    if( xMQTTStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send UNSUBSCRIBE packet to broker with error = %u.",
+                    xMQTTStatus ) );
+        returnStatus = pdFAIL;
+    }
+    else
+    {
+        LogInfo( ( "UNSUBSCRIBE sent topic %.*s to broker.\n\n",
+                   usTopicFilterLength,
+                   pcTopicFilter ) );
+
+        /* Process incoming packet from the broker. Acknowledgment for subscription
```

Comment:
Update the comment to say Unsubscribe.

Reply:
For unsubscription, I will simply the comment as it's in MQTT demo.
```
/* Process the incoming packet from the broker. */
```